--- libxml2-2.7.7/xml2-config.in.org	2010-11-12 12:02:01.000000000 +0100
+++ libxml2-2.7.7/xml2-config.in	2010-11-12 13:09:26.000000000 +0100
@@ -67,7 +67,7 @@
 	;;
 
     --cflags)
-       	echo @XML_INCLUDEDIR@ @XML_CFLAGS@
+       	pkg-config libxml-2.0 --cflags
        	;;
 
     --libtool-libs)
@@ -82,17 +82,7 @@
        	;;
 
     --libs)
-        if [ "`uname`" = "Linux" ]
-	then
-	    if [ "@XML_LIBDIR@" = "-L/usr/lib" -o "@XML_LIBDIR@" = "-L/usr/lib64" ]
-	    then
-		echo @XML_LIBS@ 
-	    else
-		echo @XML_LIBDIR@ @XML_LIBS@ 
-	    fi
-	else
-	    echo @XML_LIBDIR@ @XML_LIBS@ @WIN32_EXTRA_LIBADD@
-	fi
+       	pkg-config libxml-2.0 --libs
        	;;
 
     *)
