--- xserver-xorg-video-imx-10.07.11.orig/configure.ac	2010-10-04 17:30:52.000000000 +0200
+++ xserver-xorg-video-imx-10.07.11/configure.ac	2010-10-04 17:30:41.000000000 +0200
@@ -29,7 +29,7 @@
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_CONFIG_HEADER([config.h])
 AC_CONFIG_AUX_DIR(.)
-AC_CONFIG_MACRO_DIR([/usr/share/aclocal])
+#AC_CONFIG_MACRO_DIR([/usr/share/aclocal])
 
 AM_INIT_AUTOMAKE([dist-bzip2])
 
