diff -Naur imx-test-4.5.1_orig/test/make.rules imx-test-4.5.1/test/make.rules
--- imx-test-4.5.1_orig/test/make.rules	2009-08-06 09:39:36.000000000 +0200
+++ imx-test-4.5.1/test/make.rules	2010-06-30 13:49:57.000000000 +0200
@@ -4,15 +4,7 @@
 	VERBOSE=@
 endif
 
-AS		= $(CROSS_COMPILE)as
-LD		= $(CROSS_COMPILE)ld
-CC		= $(CROSS_COMPILE)gcc
 CPP		= $(CC) -E
-AR		= $(CROSS_COMPILE)ar
-NM		= $(CROSS_COMPILE)nm
-STRIP		= $(CROSS_COMPILE)strip
-OBJCOPY		= $(CROSS_COMPILE)objcopy
-OBJDUMP		= $(CROSS_COMPILE)objdump
 RM		= rm -f
 MAKEDIR		= mkdir -p
 
diff -Naur imx-test-4.5.1_orig/test/mxc_rng_test/Makefile imx-test-4.5.1/test/mxc_rng_test/Makefile
--- imx-test-4.5.1_orig/test/mxc_rng_test/Makefile	2009-08-06 09:39:36.000000000 +0200
+++ imx-test-4.5.1/test/mxc_rng_test/Makefile	2010-06-30 13:49:27.000000000 +0200
@@ -33,7 +33,6 @@
 # Location of local headers and API "common" sources
 INCLUDEDIR= -I. -I../../include -I../mxc_sahara_test $(INC)
 
-AR = $(CROSS_COMPILE)ar -crv
 RM = rm -f
 
 # Add compilation checks
@@ -51,8 +50,6 @@
 # In absence of kernel CONFIG flags, set API library to build on
 # 'unknown' platform which has an RNGA.
 CFLAGS += $(TARGET_ARCH)
-CC=$(CROSS_COMPILE)gcc
-LD=$(CROSS_COMPILE)ld
 
 all: $(OBJS)
 
diff -Naur imx-test-4.5.1_orig/test/mxc_vpu_test/Makefile imx-test-4.5.1/test/mxc_vpu_test/Makefile
--- imx-test-4.5.1_orig/test/mxc_vpu_test/Makefile	2009-08-06 09:39:36.000000000 +0200
+++ imx-test-4.5.1/test/mxc_vpu_test/Makefile	2010-06-30 13:50:14.000000000 +0200
@@ -1,6 +1,3 @@
-CC=$(CROSS_COMPILE)gcc
-LINK=$(CROSS_COMPILE)gcc
-
 OBJ = main.o \
       dec.o \
       enc.o \
